tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestTimeCycle.java

```diff
@@ -0,0 +1,88 @@
+package org.unicode.cldr.unittest;
+
+import com.ibm.icu.text.DateTimePatternGenerator;
+import com.ibm.icu.text.SimpleDateFormat;
+import com.ibm.icu.util.ULocale;
+import java.util.logging.Logger;
+import org.junit.jupiter.api.Test;
+import org.unicode.cldr.icu.LDMLConstants;
+import org.unicode.cldr.util.CLDRConfig;
+import org.unicode.cldr.util.CLDRLocale;
+import org.unicode.cldr.util.Factory;
+import org.unicode.cldr.util.ICUServiceBuilder;
+
+public class TestTimeCycle {
+    private static final Logger logger = Logger.getLogger(TestTimeCycle.class.getName());
+
+    /**
+     * Test that a locale's standard time format matches the region's preferred time cycle
+     *
+     * <p>This is similar to icu4j DateTimeGeneratorTest.testJjMapping. It uses current CLDR data,
+     * not ICU data.
+     */
+    @Test
+    public void testJjMapping() {
+        final String jSkeleton = "j";
+        final char[] timeCycleChars = {'H', 'h', 'K', 'k'};
+        // First test that j maps correctly by region in a locale for which we do not have data.
```

Comment:
This first part is really just a test of ICU code (and using ICU data), so there is not much point to having it in CLDR as a unit test. I would just get rid of it.